### ISSUE TITLE
TypeVar Inconsistency with mypy

### ISSUE THREAD

**Issue by alxdb** [microsoft/pyright#1617]
**Describe the bug**
A specific arrangement of TypeVar type annotations, creates an error in pyright, where there is no such error in mypy

**To Reproduce**
```python
from typing import Iterator, Optional, Sequence, TypeVar, Set

T = TypeVar("T", int, float)


def f(xs: Sequence[Optional[T]]) -> Iterator[T]:
    return (x for x in xs if x is not None)


def g(xs: Sequence[Optional[T]]) -> Iterator[T]:
    return f(xs)
```

```
$ pyright test.py
No configuration file found.
stubPath /Users/alexander.bryan/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
/Users/alexander.bryan/test.py
  /Users/alexander.bryan/test.py:11:12 - error: Expression of type "Iterator[float]" cannot be assigned to return type "Iterator[T@g]"
    TypeVar "_T_co@Iterator" is covariant
      Type "float" cannot be assigned to type "T@g" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
Completed in 0.557sec

$ mypy test.py
Success: no issues found in 1 source file
```

**Expected behavior**
no errors are reported from pyright

**VS Code extension or command-line**
cli, pyright version 1.1.120, installed via npm

**Additional context**
mypy version 0.812, installed via pip

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.121, which I just published. It will also be included in the next release of pylance.
